lib/file-store.js

```diff
@@ -380,6 +381,7 @@ const FileStore = function(rootDirectory) {
           if (err) {
             return done("Error updating metadata");
           }
+          metaData.key = destKey;
```

Comment:
@anantab Can you move this into the `createMetaData` call? It doesn't look like `key` is being used in `createMetaData`, so you can change `key: srcKey` to `key: destKey` instead to achieve the same result. This way we don't increase coupling between these callbacks and `createMetaData`, and you can delete line 339.